test/integration/test_conferences.py

```diff
@@ -0,0 +1,584 @@
+    """
+    Voice Conferences API integration test
+    """
+
+    def setUp(self):
+        """
+        Set up for our tests by creating the CallsApi and ConferencesApi instances
+        for testing as well as the unauthorized and forbidden credentials for the 4xx tests.
+        """
+        configuration = bandwidth.Configuration(
+            username = BW_USERNAME,
+            password = BW_PASSWORD
+        )
+        api_client = bandwidth.ApiClient(configuration)
+
+        # Two Valid API Clients
+        self.calls_api_instance = calls_api.CallsApi(api_client)
+        self.conference_api_instance = conferences_api.ConferencesApi(api_client)
+
+        # Unauthorized API Client
+
+        unauthorizedConfiguration = bandwidth.Configuration(
+             username='bad_username',
+             password='bad_password'
+        )
+        unauthorized_api_client = bandwidth.ApiClient(unauthorizedConfiguration)
+        self.unauthorized_api_instance = conferences_api.ConferencesApi(unauthorized_api_client)            
+        
+        # Forbidden API Client
+
+        forbiddenConfiguration = bandwidth.Configuration(
+            username=FORBIDDEN_USERNAME,
+            password=FORBIDDEN_PASSWORD
+        )
+        forbidden_api_client = bandwidth.ApiClient(forbiddenConfiguration)
+        self.forbidden_api_instance = conferences_api.ConferencesApi(forbidden_api_client)
+
+        # Rest client for interacting with Manteca
+        self.rest_client = RESTClientObject(bandwidth.Configuration.get_default_copy())    
+        configuration = bandwidth.Configuration(
+            username = BW_USERNAME,
+            password = BW_PASSWORD,
+        )
+
+        self.account_id = BW_ACCOUNT_ID
+
+    def tearDown(self):
+        """
+           Whenever we create an actual call, we'll add the call_id to the callIdArray. Then when the integration test is done, as part of tearDown we'll:
+                Do a get to check is the call status is still active
+                    If so, update to completed to end the call
+                    If not, pop that callID off the array
+                Once we go through the whole array, we clear the array so it's empty for the next integration test.    
+           if the status is active, send UpdateCall to change to completed
+        """     
+        
+        if len(callIdArray) > 0:       
+            for callId in callIdArray:
+                body = UpdateCall(state=CallStateEnum("completed"))
+                get_call_response: CallState = self.calls_api_instance.get_call_state(BW_ACCOUNT_ID, callId, _return_http_data_only=False)
+                if get_call_response[0].state == 'active':
+                    self.calls_api_instance.update_call(BW_ACCOUNT_ID, callId, body, _return_http_data_only=False)
+                elif get_call_response[0].state == 'complete':
+                    callIdArray.remove(callId)
+            callIdArray.clear()
+        pass         
```

Comment:
**_Refactoring_** This function is used in multiple Tests, opportunity to remove duplication. We can create a **base** Test Class with common functionality including this function and use where applicable for `Calls API` usage.